cloudmanager/cloudmanager.go

```diff
@@ -0,0 +1,256 @@
+package cloudmanager
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/url"
+	"runtime"
+
+	"github.com/mongodb/go-client-mongodb-atlas/mongodbatlas"
+)
+
+// Version for client
+// DefaultBaseURL API default base URL
+// DefaultUserAgent To be submitted by the client
+const (
+	Version          = "0.1"
```

Comment:
I have mixed feelings about it, I like the idea of injecting it as it would be the smarter/correct thing to do when we star packaging but I've been doing some research about this thing, go and clients and some people are just versioning with the API and following the go mod recommendation to package the module with that version, so something like api v1 would live in a v1 folder and the version of the client is that and so on but I don't like that as much as injecting but in a way feels like the "go way"

Reply:
The go thing about having a `v1` dir is mostly related, in my understanding, to libraries that existed pre-modules.  So if you had a 1.x.x without `go mod` support you could create a `v2` dir and put all `2.x.x` code there, thus supporting both mod/no-mod.

However, you wouldn't have a dir for every `x.y.z`.  Which leads to 'injecting' the version and build time.  If you build on a standard commit, you'd inject something like 7-char gitsha as the version.  If a tag exists for that sha, you inject the version.  I have code for this written somewhere, it's not public, but I can share it. 

In any case, we should solve this project in another PR, as this one is pretty big already. :)